### PR TITLE
Repository Types and ListRepositoriesResponse

### PR DESCRIPTION
**Pull request by mmacheerpuppy**
Types to support Repositories for Github, and list the repositories response

### CODE REVIEW

File: api/types/integrations_github.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	out := make([]Repository, 0, len(gh))
+
+	for _, repository := range gh {
+		if repository != nil {
+			out = append(out, NewRepository(repository))
+		}
+	}
+
+	return out
+}
```
Comment (noorvir) [inline api/types/integrations_github.go:32]: Let's move this out of types into an internal package.

File: api/types/management.go [hunk 2]
```diff
@@ -34,15 +34,16 @@ type AccessTokensDeleteResponse struct {
...
+	UserID              string    `json:"userID"`
+	Email               string    `json:"email"`
+	GithubID            int32     `json:"githubID"`
+	GithubUsername      string    `json:"githubUsername"`
+	DateJoined          time.Time `json:"dateJoined"`
+	Credit              string    `json:"credit"`
+	FirstName           string    `json:"firstName"`
+	LastName            string    `json:"lastName"`
+	Providers           []string  `json:"providers"`
+	GithubCredentialsID *string   `json:"-,omitempty"`
```
Comment (noorvir) [inline api/types/management.go:46]: What is the `GithubCredentialsID` here?

Comment (mmacheerpuppy) [reply api/types/management.go:46]: this is the address we store the credentials ID for the user, it's part of a user's `account`. no i don't think we need to expose it to the user but then we have diverging models for a user account (in infra) and the user account (in `unweave/unweave`) ; any thoughts on how we can resolve this?

Comment (noorvir) [reply api/types/management.go:46]: I think we can just use the db types to get the `GithubCredentialsID` in infra.

**Review by noorvir — APPROVED** [review]

**Pull request merged**